instill/model/v1alpha/definition.proto

```diff
@@ -2,121 +2,98 @@ syntax = "proto3";
 
 package instill.model.v1alpha;
 
+// Protocol Buffers Well-Known Types
+import "google/protobuf/struct.proto";
+
 // Protobuf standard
 import "google/protobuf/timestamp.proto";
 
 // Google api
+import "google/api/resource.proto";
 import "google/api/field_behavior.proto";
 
-///////////////////////////////
-// GitHub Model Spec represents a model spec data created from GitHub
-message Configuration {
-  // GitHub repository URL
-  string repo = 1 [ (google.api.field_behavior) = OUTPUT_ONLY ];
-  // GitHub tag
-  string tag = 2 [ (google.api.field_behavior) = OUTPUT_ONLY ];
-  // Model description
-  string html_url = 3 [ (google.api.field_behavior) = OUTPUT_ONLY ];
-}
-
-///////////////////////////////
-// Model owner
-message Owner {
-  // Owner uuid
-  string id = 1 [ (google.api.field_behavior) = OUTPUT_ONLY ];
-  // Owner username
-  string username = 2 [ (google.api.field_behavior) = OUTPUT_ONLY ];
-  // Owner type
-  string type = 3 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+////////////////////////////////////
+// Spec represents a spec data model
+message Spec {
+  // Spec documentation URL
+  string documentation_url = 1 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // Spec specification
+  google.protobuf.Struct specification = 2 [ (google.api.field_behavior) = REQUIRED ];
 }
 
 ///////////////////////////////////////////////////////////////////
 // ModelDefinition represents the definition of a model
 message ModelDefinition {
-  // Model visibility including public or private
-  enum Visibility {
-    // Visibility: UNSPECIFIED
-    VISIBILITY_UNSPECIFIED = 0;
-    // Visibility: PUBLIC
-    VISIBILITY_PUBLIC = 1;
-    // Visibility: PRIVATE
-    VISIBILITY_PRIVATE = 2;
-  }
-  // Model source
-  enum Source {
-    // Source: UNSPECIFIED
-    SOURCE_UNSPECIFIED = 0;
-    // Source: GITHUB
-    SOURCE_GITHUB = 1;
-    // Source: LOCAL
-    SOURCE_LOCAL = 2;
-  }
+  option (google.api.resource) = {
+    type: "api.instill.tech/ModelDefinition"
+    pattern: "model-definitions/{uuid}"
+  };
+
+  // ModelDefinition resource name. It must have the format of "model-definitions/{uuid}"
+  string name = 1 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // ModelDefinition UUID
+  string id = 2 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // ModelDefinition name
+  string display_name = 3 [ (google.api.field_behavior) = OUTPUT_ONLY ];
+  // ModelDefinition documentation url
+  string documentation_url = 4 [ (google.api.field_behavior) = OUTPUT_ONLY ];
```

Comment:
@xiaofei-du the Spec contain `documentation_url` and `ModelDefinition` contain a spec field. Do we need `documentation_url` in  `ModelDefinition` ?

Reply:
currently yeah. This seems duplicated, but it happens due to how the definition and spec JSON schema is designed and generated. You can refer to `connector-backend` [definitions.yaml](https://github.com/instill-ai/connector-backend/blob/main/configs/init/airbyte/seed/destination_definitions.yaml) and [specs.yaml](https://github.com/instill-ai/connector-backend/blob/main/configs/init/airbyte/seed/destination_specs.yaml).